rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -145,12 +147,8 @@ def __init__(
         self.chain_modules_prefix_length = len(self.chain_modules_root)
         self.dbevmtx = DBEvmTx(self.database)
         self.dbevents = DBHistoryEvents(self.database)
-        self.base = BaseDecoderTools(
-            database=database,
-            evm_inquirer=self.evm_inquirer,
-            is_non_conformant_erc721_fn=self._is_non_conformant_erc721,
-            address_is_exchange_fn=self._address_is_exchange,
-        )
+        self.base: BaseDecoderTools
+        self.base = base_tools
```

Comment:
This is probably not needed as it's exactly the same as the one you assign to

```suggestion
        self.base = base_tools
```